File: java_algorithm/TimeInclude3.java
import java.io.*;

public class TimeInclude3 {
    public static boolean check(int h, int m, int s) {
        if(h % 10 == 3 || m % 10 == 3 || m / 10 == 3 || s % 10 == 3 || s / 10 == 3)
            return true;

        return false;
    }
    // 완전탐색
    public static void solution() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int cnt = 0; // 3을 포함하는 시간 카운트
        int h = Integer.parseInt(br.readLine()); // 시간 입력
        br.close();
        for (int i = 0; i <= h; i++) { // h 시
            for (int j = 0; j < 60; j++) { // 분
                for (int k = 0; k < 60; k++) { // 초
                    if (check(i, j, k)) {
                        cnt++;
                    }
                }
            }
        }
        bw.write(cnt+"");
        bw.flush();
        bw.close();
    }

    public static void main(String[] args) throws IOException {
        solution();
    }
}

File: week5/TimeInclude3.cpp
#include <iostream>
using namespace std;
int h, cnt;
// 특정한 시각 안에 '3'이 포함되어 있는지의 여부
bool check(int h, int m, int s) {
	if (h % 10 == 3 || m / 10 == 3 || m % 10 == 3 || s / 10 == 3 || s % 10 == 3)
		return true;
	return false;
}
void main() {
	// H 입력 받기
	cin >> h;
	for (int i = 0; i <= h; i++) {
		for (int j = 0; j < 60; j++) {
			for (int k = 0; k < 60; k++) {
				if (check(i, j, k)) cnt++;
			}
		}
	}
	cout << cnt << '\n';
}